docs/proposals/initializers.md

```diff
@@ -0,0 +1,110 @@
+
+While powerful, admission controllers have drawbacks.  The major drawback is that it requires a recompilation and restart of the API server to introduce new behavior.  For those scenarios where an immediate yes/no response was not required for the user experience, it would be ideal if the system could support an asynchronous visitor pattern that would allow controllers external to the API server to observe, and modify a resource prior to it becoming 'active' in the system.  This would allow a new type of controller to be authored in the system that can initialize a resource similar to how other controllers watch and act upon already initialized objects.  There are many scenarios where this would be appropriate.  For example, a pod auto-sizer could be deployed separate from the API server that watches for incoming pods.  The pod-autosizer should be able to assign compute resource requests prior to the pod being scheduled.  If authored as an "initializer", it would allow anyone to bring their own logic to the cluster for how sizing occurs without needing to recompile or change the core components.
+
+## Goals
+
+1. Enable asynchronous controller pattern (i.e. initializers) as a viable alternative to existing admission control to support loose coupling of components where synchronous user response is less critical.
+2. Enable registration of a set of initializers to an API resource that must be able to act on an object to support its initialization.
+
+## Model changes
+
+**Initializer**
+
+A new type introduced into the API that allows the registration of an initializer with an associated group and kind in the API.  This allows dynamic registration of initializers without a forced restart or recompilation of the server.
+
+**NOTE: It's possible we could support this without introduction of a new type, maybe via the ConfigData resource, needs thought and input from community**
+
+**ObjectMeta.Initializers**
+
+Map of named tokens representative of external controllers that must visit a resource prior to becoming active. The set of initializers are defaulted based on the API Group and Kind of resource as part of configuration using our existing admission control facility.  For example, if an Initializer (name=initial-sizer) is associated with (group/kind=pod), then when a pod is initially created via the API, it will have its ObjectMeta.Initializers field set to "initial-sizer".
```

Comment:
See https://github.com/kubernetes/kubernetes/issues/3585#issuecomment-132855944

The reason for a map is to identify the particular initializer responsible for that category of initialization: scheduling, initial sizing, etc. That necessary in order to override the default behavior.